src/com/sap/piper/ConfigurationLoader.groovy

```diff
@@ -1,44 +1,104 @@
 package com.sap.piper
 
+// script is present in the signatures in order to keep api compatibility.
+// The script referenced is not used inside the method bodies.
+
 @API(deprecated = true)
 class ConfigurationLoader implements Serializable {
+
+    static Map stepConfiguration(String stepName) {
+        return stepConfiguration(null, stepName)
+    }
+    @Deprecated
+    /** Use stepConfiguration(stepName) instead */
     static Map stepConfiguration(script, String stepName) {
-        return loadConfiguration(script, 'steps', stepName, ConfigurationType.CUSTOM_CONFIGURATION)
+        return loadConfiguration('steps', stepName, ConfigurationType.CUSTOM_CONFIGURATION)
+    }
+
+    /*
+     * By default this methods does nothing. With this method we are able to ensure that we do not call the
+     * deprecated methods. Might be usefull during local development.
+     */
+    private static handleDeprecation(script, String methodName) {
+        if(script != null) {
```

Comment:
i think also with script==null they are using a deprecated method, as it is about with which parameters the method is called, not with which values.

Reply:
mmhhh. Maybe misunderstanding. That was basically used by me during the development process for checking if we still call some deprecated methods. I did not have in mind calling the method with script in signature with the value null. In order to avoid further discussion on that no-topic I can also remove that.